opentelemetry-semantic-conventions/scripts/generate-consts-from-spec.sh

```diff
@@ -51,16 +70,15 @@ fi
 "${SED[@]}" "s/\(opentelemetry.io\/schemas\/\)[^\"]*\"/\1$SPEC_VERSION\"/" src/lib.rs
 
 # handle doc generation failures
-"${SED[@]}" 's/\[2\]\.$//' src/resource.rs src/trace.rs # remove trailing [2] from few of the doc comments
+"${SED[@]}" 's/\[2\]\.$//' src/attribute.rs # remove trailing [2] from few of the doc comments
 
-# Remove the messaging.client_id definition along with its comments from the generated files
-#   - semconv "messaging.client_id" is deprecated
-#   - semconv "messaging.client.id" is to be used instead
-#   - Now because we use:
-#      	pub const {{attribute.fqn | to_const_name}}: &str = "{{attribute.fqn}}";
-#     to generate the consts, where to_const_name replaces '.' with '_', we need to remove the old definition
-#	  to avoid conflicts with the new one. Refer - https://github.com/open-telemetry/semantic-conventions/issues/1031
-"${SED[@]}" '/\/\/\/ Deprecated, use `messaging.client.id` instead\./{N;N;N;N;d;}' src/trace.rs src/resource.rs
-"${SED[@]}" '/pub const MESSAGING_CLIENT_ID: &str = "messaging.client_id";/{N;d;}' src/trace.rs src/resource.rs
```

Comment:
This special handling for `messaging.client_id` was to resolve the conflict as same variable was getting created for new and deprecated value. Is this no more an issue?

Reply:
Hopefully! I moved the logic to the Jinja template itself: https://github.com/open-telemetry/opentelemetry-rust/pull/1918/files#diff-1a84552ce3e0d372a5f57ad55a985d0dd2a8af846561449c804ef3d835dfae66R14-R16

If there are multiple which resolve to the same name variable name, it takes the first non-deprecated. If there aren't any, it returns the first of whatever deprecated variable is available.

Reply:
Thanks, this looks cleaner approach. Though not sure if I am missing something, as of now I only see the variable name with deprecated value in this PR i.e., 
```rust
pub const MESSAGING_CLIENT_ID: &str = "messaging.client_id";
```
while it should be:
```rust
pub const MESSAGING_CLIENT_ID: &str = "messaging.client.id";

```

Reply:
Ah poop, good catch. I'd guess `selectattr('stability_string', '!=', 'StabilityLevel.DEPRECATED')` (or something similar) isn't functioning as I expect. I'll look into fixing it when I get a chance.

Reply:
@lalitb Ah I misunderstood the potential values for StabilityLevel (looks like the original `StabilityLevel.DEPRECATED` level referenced in the script doesn't exist? https://github.com/open-telemetry/build-tools/blob/ffcffc73b3575942932de31cdae0ffe35c181186/semantic-conventions/src/opentelemetry/semconv/model/semantic_attribute.py#L40). Explains why there are no `#[deprecated]` macros anywhere. Correct key to reference seems to be `attribute.deprecated`.

Reply:
Believe this should be fixed now!

Reply:
Thanks @tbrockman yes it should have been `attribute.deprecated`. The variable seems to be created correctly now.